app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    author.update(postscounter: author.posts.count)
+  end
+
```

Comment:
__Same here__
- [ ] [OPTIONAL] Instead of calling count on records (image 1000 records, it will slow the application), we can simply use the ...increment... Ruby method. Something like this: 
```ruby
 def update_post_counter
    author.increment!(:posts_counter)
  end
```
__Since you are using an after_save callback function, this function should be a private function.__